### ISSUE TITLE
[ISSUE] Fetching translated resources type: 0

### ISSUE THREAD

**Issue by DennisKraaijeveld** [fedehub93/gatsby-source-shopify-translations#3]
Hi, I am having a hard time setting this up, and I am not exactly sure if there is any kind of support and if this plugin will be maintained or not, but multilingual functionality is really needed.. So I translated my products and collections with a third party app (Langshop) but still it wont let me fetch the translated products and collections

```
info Fetching translated resources type ShopifyTranslatedProduct: 0
info Fetching translated resources type ShopifyTranslatedCollection: 0
```

```
   {
      resolve: 'gatsby-source-shopify',
      options: {
        password: process.env.SHOPIFY_SHOP_PASSWORD,
        storeUrl: process.env.GATSBY_SHOPIFY_STORE_URL,
        shopifyConnections: ['collections'],
      },
    },
    {
      resolve: `gatsby-source-shopify-translations`,
      options: {
        shopName: process.env.GATSBY_SHOPIFY_STORE_URL,
        shopifyPassword: process.env.SHOPIFY_SHOP_PASSWORD,
        accessToken: process.env.GATSBY_STOREFRONT_ACCESS_TOKEN,
        defaultLang: 'nl',
        prefixDefault: false,
        configPath: require.resolve('./locales/config.json'),
        locales: ['nl', 'de'],
      },
    },
```

**Comment by fedehub93** [comment]
Hi Dennis, I didn't test the plugin with the third part app Langshop, but if it uses the native Translation API of shopify, it should work.

I used the Language Native Translate App to insert my translations to test.

You can test with Postman or Shopify GraphQL App the query the plugin runs to get translated resources.
Try it to check if you get data in response.

**GraphQL Query**

```
query translatedThings($ids: [ID!]!) {
  nodes(ids: $ids) {

    ... on Product {
      __typename
      title
      description
      descriptionHtml
      options {
        __typename
        name
        values
      }
    }
  }
}
```

**Variables**

```
{
  "ids": [
    "gid://shopify/Product/[XXXXXXXXXXXXX]"
  ]
}
```

EDIT: Remember you have to add these key/value on Headers correctly :

EXAMPLE URI: https://[SHOP_NAME].myshopify.com/api/2021-10/graphql.json
METHOD: POST

- X-Shopify-Storefront-Access-Token: [YOUR SHOPIFY ACCESS TOKEN]
- Accept-Language: [LANGUAGE (nl)]

**Comment by DennisKraaijeveld** [comment]
Thank you for the response! I did make a POST request with Postman, and yes, the translated content comes back perfectly. So your query is working. But the plugin is still returning zero nodes. I feel there is something wrong in my plugin settings then..

1. I deactivated the Gatsby Source Shopify Plugin which was running first
2. Have a config.json
3. Followed every step as told

The translations are there, but the plugin cant pick them up, which is really weird..

`warn The gatsby-source-shopify-translations plugin has generated no Gatsby nodes. Do you need it?`

Config.json:

```
[
  {
    "code": "nl",
    "pages": [
      {
        "originalPath": "/contact-us",
        "path": "/contact-us"
      }
    ]
  },
  {
    "code": "fr",
    "pages": [
      {
        "originalPath": "/contact-us",
        "path": "/contacts"
      }
    ]
  }
]
```

**Comment by fedehub93** [comment]
Ok, perfect

**gatsby-source-shopify** plugin is required and must be before **gatsby-source-shopify-translations**

Then, assuming you have configured the **.env** file correctly :

```
GATSBY_SHOPIFY_STORE_URL=[SHOP_NAME].myshopify.com
GATSBY_SHOPIFY_ACCESS_TOKEN=[SHOPIFY_ACCESS_TOKEN]
SHOPIFY_SHOP_PASSWORD=[SHOPIFY_SHOP_PASSWORD]
```

Can you also give me these info?

- Node version
- gatsby version
- gatsby-source-shopify version

Thank you so much

**Comment by DennisKraaijeveld** [comment]
Ran it one more time and double checked all my .env variables, and those are fine, but still zero translated products found. I can setup a test repo with the same environment, and make a new private app in my store for testing purposes. The translations are found when making a POST in Postman, so the translations are really there.

Thanks again! This plugin could have amazing potential, since it is something a lot of users need and there isn't anything published yet which can pull translated products to Gatsby. Great job.

Node: 14.9.0
Gatsby CLI version: 4.10.0
Gatsby version: 4.8.0
Gatsby Source Shopify: 6.10.2


I am setting up a reproduction repo right now. 

Ahw, Shopify has a new way of creating apps. So I can't make a new one since I don't know if those new private apps are working with the current Source Shopify Plugin.

**Comment by DennisKraaijeveld** [comment]
https://github.com/DennisKraaijeveld/gatsby-shopify-translation-issue 

Made a repo for my environment. I guess you could try it out with your own Shopify tokens, but you could try mine, but those are really private so I can't just post them right here.

**Comment by fedehub93** [comment]
I managed to replicate your issue with my shopify token too.

At first glance it seems that gatsby-source-shopify works asynchronously in the gatsby plugin stack. Before he finishes and creates the nodes that are needed for gatsby-source-shopify-translations, this last plugin is already working and doesn't find anything.

I think the problem is based on the large amount of nodes that are created too.

I'd never had this problem before but I'll try to solve it.

**Comment by DennisKraaijeveld** [comment]
> I managed to replicate your issue with my shopify token too.
> 
> 
> 
> At first glance it seems that gatsby-source-shopify works asynchronously in the gatsby plugin stack. Before he finishes and creates the nodes that are needed for gatsby-source-shopify-translations, this last plugin is already working and doesn't find anything.
> 
> 
> 
> I think the problem is based on the large amount of nodes that are created too.
> 
> 
> 
> I'd never had this problem before but I'll try to solve it.

Thanks! Looking forward to a fix :) I am gonna ask around for you and will come back at you for a possible solution 

One question. I think it should be possible to use this plugin to create multilingual websites like the following:

domain.de: query for the default lang and 'de' language, but only use the translated content to create the collection and product pages
domain.fr: query for the default lang and 'fr' language, but only use the translated content to create the collection and product pages

Because I like the setup to have multiple domains (for SEO purposes) instead of having: domain.com/fr/collection

**Comment by fedehub93** [comment]
I released a new version 0.3.0, so you can try again.

At the moment it is not possible to manage with multiple domains although it was something I had already thought about.

Since shopify has released the new Storefront API of 2022-01 which finally also concern translations, I wanted to rethink the plugin from this perspective that would make everything simpler and more functional. 

Maybe it could also be time to expand translations by domain as well.

**Comment by DennisKraaijeveld** [comment]
Thanks! I think it would work, only now it is hitting Shopify rate limits (I think). The error I did get:

```
"gatsby-source-shopify-translations" threw an error while running the sourceNodes lifecycle:

Input size greater than 250.
Error: Input size greater than 250.
  
  - index.js:29 new ApolloError
    [vivhome]/[@apollo]/client/errors/index.js:29:28
  
  - QueryManager.js:592 
    [vivhome]/[@apollo]/client/core/QueryManager.js:592:47
```

Shopify Resource:

Maximum input array size limit [Anchor link to section titled "Maximum input array size limit"](https://shopify.dev/api/usage/rate-limits#maximum-input-array-size-limit)
Input arguments that accept an array have a maximum size of 250. Queries and mutations return an error if an input array exceeds 250 items.

https://shopify.dev/api/usage/bulk-operations/queries


I have around 1070 products btw

**Comment by fedehub93** [comment]
I released the 0.3.1 to fix the Maximum input array size limit error.

You can test and see how it works in your context and with your number of products.
In case everything goes well, I close this issue.

**Comment by DennisKraaijeveld** [comment]
Everything went well. Thanks! Now I am going to figure out how to setup things and see if I can somehow only query a second language for a second storefront.
